pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -1372,6 +1380,64 @@ async def send_image_list_email(self, release_name: str, advisory: int, archive_
         subject = f"OCP {release_name} Image List"
         return await exectools.to_thread(self._mail.send_mail, self.runtime.config["email"]["promote_image_list_recipients"], subject, content, archive_dir=archive_dir, dry_run=self.runtime.dry_run)
 
+    def send_notification_email(self, group_config, release_name, impetus_advisories):
+        """
+        Send a notification email to QEs if it hasn't been done yet
+        check release jira subtask for task status
+        """
+
+        jira_issue_key = group_config.get("release_jira")
+        if not jira_issue_key:
+            return
+
+        self._logger.info("Checking notify QE release subtask")
+        parent_jira = self._jira_client.get_issue(jira_issue_key)
+        title = "Notify QE of release advisories"
+        subtask = next((s for s in parent_jira.fields.subtasks if title in s.fields.summary), None)
+        if not subtask:
+            raise ValueError("Notify QE release subtask not found in release_jira: %s", jira_issue_key)
+
+        if subtask.fields.status.name == "Closed":
+            return
+
+        self._logger.info("Sending a notification to QE and multi-arch QE...")
+        jira_issue_link = "https://jira.example.com/browse/FOO-1" if self.dry_run else parent_jira.permalink()
+        nightlies = group_config.get("reference_releases", {})
+        nightlies_w_pullspecs = nightlies_with_pullspecs(nightlies.values())
+        self._send_notification_email(release_name, impetus_advisories, jira_issue_link,
```

Comment:
Should this run if `self.dry_run` is True?

Reply:
yeah the func accounts and logs the email content in the job. To see it in action: https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/sidsharm-aos-cd-jobs/job/build%252Fpromote-assembly/19/console